src/Web/Intercom/Types.hs

```diff
@@ -50,23 +52,40 @@ ping c = do
 opts c = defaults & auth ?~ basicAuth (c ^. appId) (c ^. apiKey)
                   & header "Accept" .~ ["application/json"]
 
+postOpts c = opts c & header "Content-Type" .~ ["application/json"]
+
 data User = User {
   _name             :: Maybe Text,
   _email            :: Maybe Text,
   _userId           :: Maybe Text,
-  _customAttributes :: Map String Value
+  _customAttributes :: HashMap Text Value
 } deriving (Eq, Show)
 
+-- | A blank user, for overriding
+-- > let myUser = blankUser & email .~ (Just "bob@foo.com")
+blankUser :: User
+blankUser = User Nothing Nothing Nothing (fromList [])
+
 $(makeLenses ''User)
 
 instance FromJSON User where
-  parseJSON (Object v) = do
+  parseJSON (Object v) =
     User <$> v .:? "name"
     <*> v .:? "email"
     <*> v .:? "user_id"
     <*> v .: "custom_attributes"
   parseJSON _ = mzero
 
+instance ToJSON User where
+  toJSON (User n e u c) = object (nonEmpty attributes)
+    where nonEmpty = catMaybes . fmap sequenceA
```

Comment:
the attributes have the format `("name", Just "bob")`, `sequenceA` pulls out the `Maybe` functor from the second item in the tuple: `Just ("name", "bob")`. `fmap` applies it to the whole list:

`[Just ("name", "bob"), Just ("email", "bob@foo.com"), Nothing, Just ("custom_attributes", Object ...)]`